flink-connectors/flink-connector-kafka/src/main/java/org/apache/flink/connector/kafka/source/KafkaSource.java

```diff
@@ -120,7 +121,16 @@ public Boundedness getBoundedness() {
         Supplier<KafkaPartitionSplitReader<OUT>> splitReaderSupplier =
                 () ->
                         new KafkaPartitionSplitReader<>(
-                                props, deserializationSchema, readerContext.getIndexOfSubtask());
+                                props,
+                                deserializationSchema,
+                                readerContext.getIndexOfSubtask(),
+                                ((OperatorMetricGroup) readerContext.metricGroup())
```

Comment:
This cast will fail in all tests of [KafkaSourceReaderTest](https://github.com/AHeise/flink/blob/bcbc8a4d713daa06c539cad3e9d72832e15e5c97/flink-connectors/flink-connector-kafka/src/test/java/org/apache/flink/connector/kafka/source/reader/KafkaSourceReaderTest.java#L61-L61) because
[TestingReaderContext](https://github.com/AHeise/flink/blob/bcbc8a4d713daa06c539cad3e9d72832e15e5c97/flink-test-utils-parent/flink-connector-test-utils/src/main/java/org/apache/flink/connector/testutils/source/reader/TestingReaderContext.java#L33-L33) uses UnregisteredMetricsGroup. Getting an ad-hoc `OperatorMetricGroup` would require adding quite a bit of dependencies here.

Reply:
Why couldn't it just an [UnregisteredOperatorMetricGroup](https://github.com/apache/flink/blob/master/flink-runtime/src/main/java/org/apache/flink/runtime/metrics/groups/UnregisteredMetricGroups.java#L197)?

Reply:
Thanks for the pointer. So I guess adding runtime dependency to `flink-connector-test-utils` is not an issue for you.

Reply:
If by "runtime dependency" you mean
```
		<dependency>
			<groupId>org.apache.flink</groupId>
			<artifactId>flink-connector-test-utils</artifactId>
			<version>${project.version}</version>
			<scope>test</scope>
		</dependency>
```
then of course I don't mind it?

Reply:
but I am wondering why you need it?

Reply:
ah now I get it, you need flink-connector-test-utils to depend on flink-runtime...

Reply:
well that should be fine, or we create a new version of the UnregisteredOperatorMetricGroup in flink-metrics-core (probably the better option).

Reply:
(We could still add the new OperatorMetricGroup interface, and just skip the wiring)

Reply:
But why?
Add the interface to flink-metrics-core, have the flink-runtime OperatorMetricGroup implement it, and then extend the UnregisteredMetricsGroup in flink-metrics-core to also implement it.
What am I missing that makes it more complicated?

Reply:
Ah I was assuming you meant the `OperatorMetricGroup` class and not the interface. I was reading the comment in braces as a different option and not the implementation of the former description.
So to summarize, we add the interface to avoid the additional dependency but resort to cast as is (just against the interface and not the class). Do we want to rename the class or have a different name for the interface?

Reply:
I think for the backport using the same name is fine I suppose.